src/main/java/seedu/exercise/logic/commands/ClearCommand.java

```diff
@@ -20,7 +21,7 @@
     private EventPayload<ReadOnlyResourceBook<Exercise>> eventPayload;
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
Because I shifted the line `EventFactory.commandToEvent(command)` from a try-catch block inside the implementation of `EventHistory.getInstance().addCommandToUndoStack(this)`. Still thinking on how to improve this. It is technically impossible for an Exception to be thrown here actually because of the way it is designed.

Reply:
Noted. document it well.

Reply:
Okay